### PR TITLE
Spark: Remove Spark 2.4

### PR DESCRIPTION
**Pull request by Fokko**

### CODE REVIEW

File: .github/workflows/spark-ci.yml [hunk 1]
```diff
@@ -73,7 +73,6 @@ jobs:
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
         restore-keys: ${{ runner.os }}-gradle-
     - run: echo -e "$(ip addr show eth0 | grep "inet\b" | awk '{print $2}' | cut -d/ -f1)\t$(hostname -f) $(hostname -s)" | sudo tee -a /etc/hosts
-    - run: ./gradlew -DsparkVersions=2.4 -DhiveVersions= -DflinkVersions= :iceberg-spark:check :iceberg-spark:iceberg-spark-2.4:check :iceberg-spark:iceberg-spark-runtime-2.4:check -Pquick=true -x javadoc
```
Comment (aokolnychyi) [inline .github/workflows/spark-ci.yml:76]: Shall we drop `spark2-tests` completely?

File: docs/spark-queries.md [hunk 2]
```diff
@@ -31,17 +31,17 @@ To use Iceberg in Spark, first configure [Spark catalogs](../spark-configuration
...
-| [DataFrame reads](#querying-with-dataframes)     | ✔️        | ✔️          |                                                |
-| [Metadata table `SELECT`](#inspecting-tables)    | ✔️        |            |                                                |
-| [History metadata table](#history)               | ✔️        | ✔️          |                                                |
-| [Snapshots metadata table](#snapshots)           | ✔️        | ✔️          |                                                |
-| [Files metadata table](#files)                   | ✔️        | ✔️          |                                                |
-| [Manifests metadata table](#manifests)           | ✔️        | ✔️          |                                                |
-| [Partitions metadata table](#partitions)         | ✔️        | ✔️          |                                                |
-| [All metadata tables](#all-metadata-tables)      | ✔️        | ✔️          |                                                |
+| Feature support                                  | Spark 3 |
```
Comment (aokolnychyi) [inline docs/spark-queries.md:34]: I am not sure how much valuable this table is given that we only have Spark 3.
@RussellSpitzer @rdblue, thoughts?

**Review by aokolnychyi — APPROVED** [review]
Thanks, @Fokko!

**Review by nastra — APPROVED** [review]

File: docs/spark-queries.md (hunk 2 above)
Comment (RussellSpitzer) [reply docs/spark-queries.md:34]: +1 Drop table

Comment (RussellSpitzer) [reply docs/spark-queries.md:34]: And remove the line SparkDSv2 is an evolving API unless we are going to note the differences between 3.x versions

Comment (aokolnychyi) [reply docs/spark-queries.md:34]: +1

File: docs/spark-ddl.md [hunk 3]
```diff
@@ -256,18 +252,6 @@ ADD COLUMN points.value.b int
 
 Note: Altering a map 'key' column by adding columns is not allowed. Only map values can be updated.
 
-In Spark 2.4.4 and later, you can add columns in any position by adding `FIRST` or `AFTER` clauses:
```
[the remark this answers is not in the document]
Comment (Fokko) [reply docs/spark-ddl.md:259]: Ah, yes!

File: docs/spark-queries.md (hunk 2 above)
Comment (Fokko) [reply docs/spark-queries.md:34]: I was also considering this, let's remove the table!

**Pull request merged**